gsy_e_sdk/setups/Grid_Operator_API_scripts/redis_grid_fees.py

```diff
@@ -0,0 +1,78 @@
+# flake8: noqa
+# pylint: disable=duplicate-code
+
+"""
+Template file for markets management through the gsy-e-sdk api client using Rest.
+"""
+import os
+from time import sleep
+from gsy_e_sdk.redis_aggregator import RedisAggregator
+from gsy_e_sdk.redis_market import RedisMarketClient
+from gsy_e_sdk.utils import log_grid_fees_information
+
+current_dir = os.path.dirname(__file__)
+
+MARKET_NAMES = [
+    "Grid",
+    "Community",
+]
+ORACLE_NAME = "dso"
+SLOT_LENGTH = 15  # leave as is
+
+
+class Oracle(RedisAggregator):
+    """Class to represent the Grid Operator client type."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+
+    def on_market_cycle(self, market_info):
+        current_market_fee = {}
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if area_dict["area_name"] in MARKET_NAMES:
+                self.add_to_batch_commands.last_market_dso_stats(area_uuid)
+                current_market_fee[area_dict["area_name"]] = area_dict[
+                    "current_market_fee"
+                ]
+        next_market_fee = self.set_new_market_fee()
+        log_grid_fees_information(MARKET_NAMES, current_market_fee, next_market_fee)
+
+    def set_new_market_fee(self):
+        """Return the market fees for each market for the next time slot."""
+        next_market_fee = {}
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if area_dict["area_name"] in MARKET_NAMES:
+                next_market_fee[area_dict["area_name"]] = 10
+
+                self.add_to_batch_commands.grid_fees(
+                    area_uuid=area_uuid,
+                    fee_cents_kwh=next_market_fee[area_dict["area_name"]],
+                )
+        self.execute_batch_commands()
+        return next_market_fee
+
+    def on_event_or_response(self, message):
+        pass
+
+    def on_finish(self, finish_info):
+        self.is_finished = True
+
+
+MarketClient = RedisMarketClient
+aggregator = Oracle(aggregator_name=ORACLE_NAME)
+
+print()
+print("Connecting to markets ...")
+
+for i in MARKET_NAMES:
+    market_registered = RedisMarketClient(area_id=i)
+    market_registered.select_aggregator(aggregator.aggregator_uuid)
+    print("----> Connected to ", i)
+    sleep(0.3)
```

Comment:
Why do you need the sleep here?

Reply:
I have investigated and here, as well as in the other scripts, the sleep is used to reduce the computation of the script. 
Does it make sense?

Reply:
No, [this](https://github.com/gridsingularity/gsy-e-sdk/pull/226/files/c4b99c13deb0ef4829d40fe253f2357c982b9005#diff-932b46af552785d7ba0048aa69e4c1e85a5379ccc2ef73ef116f677f84528cfdR78) line is responsible to not check for the finished state every milisecond which reduces the computational effort. 

The line we are talking about doe snot make sense IMO. You are waiting here for no reason AFAIU.

Reply:
Right. I will remove it.